src/code/z_play.c

```diff
@@ -1824,7 +1824,7 @@ void Gameplay_TriggerRespawn(GlobalContext* globalCtx) {
 }
 
 s32 func_800C0CB8(GlobalContext* globalCtx) {
-    return (globalCtx->roomCtx.curRoom.mesh->polygon.type != 1) && (YREG(15) != 0x20) && (YREG(15) != 0x30) &&
+    return (globalCtx->roomCtx.curRoom.meshHeader->base.type != 1) && (YREG(15) != 0x20) && (YREG(15) != 0x30) &&
```

Comment:
Please be more pedantic about UB stuff because I'm on the "if it works it works" wagon

And how to improve the situation, I don't see what can be done

Reply:
Hmm, apparently this is allowed: https://stackoverflow.com/a/11906997 . I think I'm thinking of something else regarding unions, but if the members have a common initial segment you can access it in a well-defined way, so false alarm.